File: app/src/main/java/io/github/projectbukirin/smartrice/Diseases.java
package io.github.projectbukirin.smartrice;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ImageButton;
import android.widget.ListView;

public class Diseases extends AppCompatActivity {

    ListView list;
    String[] diseasename = {"Bacterial Blight","Rice Blast","Sheath Blight","Tungro"};
    String[] desc= {"Disease","Disease","Disease","Disease"};
    Integer[] imgid={R.drawable.bacterialblight,R.drawable.riceblast,R.drawable.sheathblight,R.drawable.tungro};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_diseases);

        list = (ListView) findViewById(R.id.disease_list);
        CustomListView customListview=new CustomListView(this, diseasename,desc,imgid);
        list.setAdapter(customListview);

        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                switch (position){
                    case 0:
                        Intent intent = new Intent();
                        intent.setClass(Diseases.this, DiseasesDefinition.class);
                        startActivity(intent);
                        break;
                    case 1:
                        Intent intent1 = new Intent();
                        intent1.setClass(Diseases.this, DiseasesDefinition.DiseaseDefinition2.class);
                        startActivity(intent1);
                        break;
                    case 2:
                        Intent intent2 = new Intent();
                        intent2.setClass(Diseases.this, DiseasesDefinition.DiseaseDefinition3.class);
                        startActivity(intent2);
                        break;
                    case 3:
                        Intent intent3 = new Intent();
                        intent3.setClass(Diseases.this, DiseasesDefinition.DiseaseDefinition4.class);
                        startActivity(intent3);
                        break;
                }
            }
        });



        ImageButton androidImageButton = (ImageButton) findViewById(R.id.imageButton3);
        androidImageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                backActivity();
            }
        });

    }
    public void backActivity(){
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
    @Override
    public void onBackPressed(){
        Intent intent = new Intent(Diseases.this, MainActivity.class );
        startActivity(intent);
    }
}

File: app/src/main/java/io/github/projectbukirin/palaylab/Diseases.kt
package io.github.projectbukirin.palaylab

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ImageButton
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import io.github.projectbukirin.palaylab.DiseasesDefinition.*

class Diseases : AppCompatActivity() {
    var list: ListView? = null
    var diseasename: Array<String?> = arrayOf("Bacterial Leaf Blight", "Rice Blast", "Sheath Blight", "Tungro")
    var desc: Array<String?> = arrayOf("Disease", "Disease", "Disease", "Disease")
    var imgid: Array<Int?> = arrayOf(R.drawable.bacterialblight, R.drawable.riceblast, R.drawable.sheathblight, R.drawable.tungro)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_diseases)
        list = findViewById<View>(R.id.disease_list) as ListView
        val customListview = CustomListView(this, diseasename, desc, imgid)
        list!!.adapter = customListview
        list!!.onItemClickListener = OnItemClickListener { _, _, position, _ ->
            when (position) {
                0 -> {
                    val intent = Intent()
                    intent.setClass(this@Diseases, DiseasesDefinition::class.java)
                    startActivity(intent)
                }
                1 -> {
                    val intent1 = Intent()
                    intent1.setClass(this@Diseases, DiseaseDefinition2::class.java)
                    startActivity(intent1)
                }
                2 -> {
                    val intent2 = Intent()
                    intent2.setClass(this@Diseases, DiseaseDefinition3::class.java)
                    startActivity(intent2)
                }
                3 -> {
                    val intent3 = Intent()
                    intent3.setClass(this@Diseases, DiseaseDefinition4::class.java)
                    startActivity(intent3)
                }
            }
        }
        val androidImageButton = findViewById<View>(R.id.imageButton3) as ImageButton
        androidImageButton.setOnClickListener { backActivity() }
    }

    fun backActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }

    override fun onBackPressed() {
        val intent = Intent(this@Diseases, MainActivity::class.java)
        startActivity(intent)
    }
}